### PR TITLE
upd(local): add error messages for missing resources / collections on meroxa apps run

### PR DESCRIPTION
**Pull request by jayjayjpg**
Closes https://github.com/meroxa/turbine-js/issues/241

This provides more detailed error messaging if resources or fixture tables are missing when running `meroxa apps run` on a turbine-js app.

## Blockers

- [x] add changeset
- [x] add tests
- [x] add demo to PR

## Before 

**Missing resource in app.json**

```bash
./meroxa apps run --path ../demos/jj-oracle-to-redshifts3snowflake/

Error: node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Error: ENOENT: no such file or directory, open '/Users/jj/go/src/github.com/meroxa/demos/jj-oracle-to-redshifts3snowflake/undefined'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/jj/go/src/github.com/meroxa/demos/jj-oracle-to-redshifts3snowflake/undefined'
}
```

**Wrong collection name in fixture**

```bash
$ ./meroxa apps run --path ../demos/jj-oracle-to-redshifts3snowflake/

Error: Error running Turbine Data app locally : Cannot read properties of undefined (reading 'forEach')
```

### New

**Missing resource in app.json**

```bash
$ ./meroxa apps run --path ../demos/jj-oracle-to-redshifts3snowflake/

Error: Error running Turbine Data app locally : You are trying to run your Turbine Data app with the resource "oracle-march15" which currently does not have a local fixture defined in your app config.
Please define a suitable fixture file for "oracle-march15" in /Users/jj/go/src/github.com/meroxa/demos/jj-oracle-to-redshifts3snowflake/app.json before running the meroxa apps run command again.
```

**Wrong collection name in fixture**

```bash
$ ./meroxa apps run --path ../demos/jj-oracle-to-redshifts3snowflake/

Error: Error running Turbine Data app locally : You are trying to read from a resource "oracle-march15" with the collection "orders" but the local fixture that you have defined for this resource in app.json does not contain a table named "orders".
Please define another fixture file for "oracle-march15" in your app.json or update the table name in your currently used fixture from "collection_name" to "orders".
```

### Default

errorme_normal

### Wrong Fixture File

errorme_wrongresource

### Wrong Table in Fixture File Defined

errorme_wrongtable

### CODE REVIEW

File: packages/turbine-js-cli/src/runtime/local.ts [hunk 1]
```diff
@@ -11,16 +12,26 @@ import { readFile } from "fs/promises";
...
     const fixturesPath = resources[resourceName];
+
+    if ((!fixturesPath || fixturesPath === "undefined") && !this.hasSource) {
+      throw new BaseError(
+        `You are trying to run your Turbine Data app with the resource "${resourceName}" which currently does not have a local fixture defined in your app config.\nPlease define a suitable fixture file for "${resourceName}" in ${this.pathToApp}/app.json before running the meroxa apps run command again.\n`
+      );
+    }
+
     const resourceFixturesPath = `${this.pathToApp}/${fixturesPath}`;
+    this.hasSource = true;
```
Comment (jayjayjpg) [inline packages/turbine-js-cli/src/runtime/local.ts:34]: With this flag I'm trying to distinguish between a `source` resource and a `destination` when calling `resources()`: only if the resource is a `source` we check for the fixture failure and if it is missing, raise an error to the console

File: packages/turbine-js-cli/src/runtime/local.ts [hunk 2]
```diff
@@ -11,16 +12,26 @@ import { readFile } from "fs/promises";
...
+    this.hasSource = false;
   }
 
   resources(resourceName: string): LocalResource {
     const resources = this.appConfig.resources;
     const fixturesPath = resources[resourceName];
+
+    if ((!fixturesPath || fixturesPath === "undefined") && !this.hasSource) {
+      throw new BaseError(
+        `You are trying to run your Turbine Data app with the resource "${resourceName}" which currently does not have a local fixture defined in your app config.\nPlease define a suitable fixture file for "${resourceName}" in ${this.pathToApp}/app.json before running the meroxa apps run command again.\n`
```
Comment (janelletavares) [inline packages/turbine-js-cli/src/runtime/local.ts:29]:
```suggestion
        `You are trying to run your Turbine Data app with the resource "${resourceName}" which currently does not have a corresponding local fixture assigned in your app config.\nPlease assign a suitable fixture file for "${resourceName}" in ${this.pathToApp}/app.json before running the meroxa apps run command again.\n`
```
nit

**Review by janelletavares — APPROVED** [review]
fantastic

File: packages/turbine-js-cli/src/runtime/local.ts (hunk 2 above)
Comment (jayjayjpg) [reply packages/turbine-js-cli/src/runtime/local.ts:29]: This reads nice, thank you! :sparkles:

**Pull request merged**